refactor: streamline component imports and enhance footer service usage

Ashwin-op opened the pull request:
# PR Template

## Type

- ( ) Fix
- ( ) Refactor
- ( ) Style
- ( ) Docs
- (X) Feat
- ( ) Hotfix
- ( ) Merge

## Description

## Screenshots

## Testing

## Bundle Size

Ashwin-op committed:
refactor: streamline component imports and enhance footer service usage

Ashwin-op committed:
fix: update component names in tests for consistency

Ashwin-op committed:
Merge branch 'test-server' of https://github.com/commudle/commudle-ng into fix-memory

Ashwin-op committed:
refactor: update communities components for better modularity and performance

Ashwin-op committed:
refactor: optimize subscriptions with takeUntil for better memory management

ArshdeepGrover commented on apps/commudle-admin/src/app/components/communities/communities-list/communities-list.component.ts:
```diff
@@ -168,16 +193,16 @@ export class CommunitiesListComponent implements OnInit, OnDestroy {
+    const queryParams: { [key: string]: string } = {};
     if (newest_communities) {
-      queryParams.newest_communities = true;
+      queryParams.newest_communities = 'true';
```
What was the purpose of shifting boolean to string?

Ashwin-op replied:
We switched to strings for query parameters because URLs only support stringsm `true` and `false` become `"true"` and `"false"` anyway. Angular's `ActivatedRoute` always returns strings, and `URLSearchParams` expects them. Using strings avoids type unions like `string | boolean` and `any`, keeps intent clear, and aligns with how the browser and Angular treat query params.

ArshdeepGrover commented on apps/commudle-admin/src/app/components/communities/communities.component.ts:
```diff
@@ -1,13 +1,37 @@
+import { RouterModule } from '@angular/router';
+import { NbIconModule } from '@commudle/theme';
 import { FooterService } from 'apps/commudle-admin/src/app/services/footer.service';
+import { AppSharedComponentsModule } from '../../app-shared-components/app-shared-components.module';
+import { ListingPageHeaderComponent } from '../../app-shared-components/listing-page-header/listing-page-header.component';
+import { ListingPagesLayoutComponent } from '../../app-shared-components/listing-pages-layout/listing-pages-layout.component';
+import { PublicHomeListEventsModule } from '../../feature-modules/listing-pages/public-home-list-events/public-home-list-events.module';
+import { PublicHomeListSpeakersModule } from '../../feature-modules/listing-pages/public-home-list-speakers/public-home-list-speakers.module';
+import { CommunitiesFeaturedComponent } from './communities-featured/communities-featured.component';
+import { CommunitiesListComponent } from './communities-list/communities-list.component';
+import { CommonModule } from '@angular/common';
```
Let's use an absolute path!

Ashwin-op replied:
done

ArshdeepGrover commented on apps/commudle-admin/src/app/components/communities/communities.component.ts:
```diff
@@ -1,13 +1,37 @@
+    CommunitiesFeaturedComponent,
+    CommunitiesListComponent,
+  ],
 })
```
Let's shift this to featured modules!

Ashwin-op replied:
done

ArshdeepGrover commented on apps/commudle-admin/src/app/components/communities/communities-list/communities-list.component.ts:
```diff
@@ -1,18 +1,41 @@
+    SharedDirectivesModule,
+    CommunitiesCardComponent,
+    SharedComponentsModule,
+  ],
```
Let's shift this to featured modules!

Ashwin-op replied:
done

ArshdeepGrover commented on apps/commudle-admin/src/app/components/communities/communities-list/communities-list.component.ts:
```diff
@@ -1,18 +1,41 @@
+import { ICommunity } from '@commudle/shared-models';
+import { IPageInfo } from '@commudle/shared-models';
+import { SeoService } from '@commudle/shared-services';
 import { Subscription } from 'rxjs';
```
Let's use absolute paths

Ashwin-op replied:
done

Ashwin-op committed:
Merge branch 'test-server' of https://github.com/commudle/commudle-ng into fix-memory

Ashwin-op committed:
feat: refactor communities module and components

- Removed the CommunitiesComponent from the main app module and created a dedicated CommunitiesModule with routing.
- Implemented lazy loading for the communities feature.
- Created CommunitiesFeaturedComponent and CommunitiesListComponent with their respective templates and styles.
- Added search functionality in CommunitiesListComponent.
- Updated the layout in CommunitiesComponent to accommodate mobile and desktop views.
- Added unit tests for the new components.

Ashwin-op committed:
refactor: update import paths for AgenciesComponent and PublicPagesRoutingModule

Ashwin-op committed:
Merge branch 'test-server' of https://github.com/commudle/commudle-ng into fix-memory

ArshdeepGrover commented on apps/commudle-admin/src/app/feature-modules/communities/components/communities/communities-list/communities-list.component.ts:
```diff
@@ -168,16 +168,16 @@ export class CommunitiesListComponent implements OnInit, OnDestroy {
+    const queryParams: { [key: string]: string } = {};
     if (newest_communities) {
-      queryParams.newest_communities = true;
+      queryParams.newest_communities = 'true';
     }
     if (members_count) {
-      queryParams.members_count = true;
+      queryParams.members_count = 'true';
     }
 
     if (completed_events_count) {
-      queryParams.completed_events_count = true;
+      queryParams.completed_events_count = 'true';
```
@Ashwin-op, is there any specific reason to change the boolean to a string?
Should it create any impact on API's?

Ashwin-op replied:
https://github.com/commudle/commudle-ng/pull/1885#discussion_r2153576949

Ashwin-op committed:
Merge branch `commudle:test-server` into fix-memory
